File: server/routes/analytics.js
const express = require('express');
const analyticsService = require('../services/analyticsService');

const router = express.Router();

router.get('/summary', async (req, res, next) => {
  try {
    res.json({ success: true, data: await analyticsService.summary(), meta: {} });
  } catch (error) {
    next(error);
  }
});

router.get('/heatmap', async (req, res, next) => {
  try {
    const data = await analyticsService.heatmap();
    res.json({ success: true, data, meta: { total: data.length } });
  } catch (error) {
    next(error);
  }
});

module.exports = router;

File: server/handlers/analytics.go
package handlers

import (
	"net/http"

	"github.com/go-chi/chi/v5"
	appmw "github.com/your-org/reelqueue-server/middleware"
	"github.com/your-org/reelqueue-server/services"
)

// AnalyticsHandler exposes analytics endpoints.
type AnalyticsHandler struct {
	analytics *services.AnalyticsService
}

// NewAnalyticsHandler creates an analytics handler.
func NewAnalyticsHandler(analytics *services.AnalyticsService) *AnalyticsHandler {
	return &AnalyticsHandler{analytics: analytics}
}

// Routes registers analytics routes.
func (h *AnalyticsHandler) Routes(r chi.Router) {
	r.Get("/summary", h.Summary)
	r.Get("/heatmap", h.Heatmap)
}

// Summary handles GET /api/analytics/summary.
func (h *AnalyticsHandler) Summary(w http.ResponseWriter, r *http.Request) {
	summary, err := h.analytics.GetSummary(r.Context())
	if err != nil {
		appmw.HandleError(w, err)
		return
	}
	appmw.Respond(w, http.StatusOK, summary, map[string]any{})
}

// Heatmap handles GET /api/analytics/heatmap.
func (h *AnalyticsHandler) Heatmap(w http.ResponseWriter, r *http.Request) {
	entries, err := h.analytics.GetHeatmap(r.Context())
	if err != nil {
		appmw.HandleError(w, err)
		return
	}
	appmw.Respond(w, http.StatusOK, entries, appmw.Meta{Total: len(entries)})
}
